src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/CitationSourceLocation.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents a location within source material used for an annotation.</summary>
+public class CitationSourceLocation
+{
+    /// <summary>Gets or sets the page at which the cited material begins.</summary>
+    public int? PageStart { get; set; }
+
+    /// <summary>Gets or sets the page at which the cited material ends.</summary>
+    public int? PageEnd { get; set; }
+
+    /// <summary>Gets or sets the block or chunk at which the cited material begins.</summary>
+    public int? BlockStart { get; set; }
+
+    /// <summary>Gets or sets the block or chunk at which the cited material ends.</summary>
+    public int? BlockEnd { get; set; }
+
+    /// <summary>Gets or sets the character at which the cited material begins.</summary>
+    public int? CharacterStart { get; set; }
+
+    /// <summary>Gets or sets the character at which the cited material ends.</summary>
+    public int? CharacterEnd { get; set; }
+}
```

Comment:
This is a tricky area. I know that if this is the data the underlying providers provide, we have to expose it in the same shape. But I feel for the poor app devs who have to somehow make a UI function on top of it - that's a lot of different combinations of data that might/might not be provided, and hard to interpret that info (e.g., what actually is a block, will the set of provided numbers change over time, etc.)

Do you think there will be enough consistency across providers to make this useful? If not I guess another option is to rely on people using `AIAnnotation.RawRepresentation` to get more provider-specific info.